distributions/aws/aws-telemetry/cronjob.yaml

```diff
@@ -0,0 +1,84 @@
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: aws-kubeflow-telemetry
+spec:
+  schedule: "0 0 * * *"
+  concurrencyPolicy: Forbid
+  successfulJobsHistoryLimit: 3
+  failedJobsHistoryLimit: 0
+  jobTemplate:
+    spec:
+      ttlSecondsAfterFinished: 0
+      backoffLimit: 3
+      template:
+        metadata:
+          annotations:
+            # istio sidecar is not neeeded since there is no inbound or outbound traffic from the service mesh
+            sidecar.istio.io/inject: "false"
+        spec:
+          restartPolicy: Never
+          containers:
+          - name: amazonlinux
+            image: public.ecr.aws/amazonlinux/amazonlinux:2
+            command:
+              - /bin/sh
+              - -c
+              - |
+                # Following code uses IMDS service. See: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html
+
+                get_instance_id() {
+                  # use IMDSv2 if enabled else fallback to IMDSv1
+                  local _token
+                  _token=$(curl -s --retry 3 --max-time 3 -X PUT http://169.254.169.254/latest/api/token -H "X-aws-ec2-metadata-token-ttl-seconds: 21600")
+                  if [[ -n ${_token+x} ]]; then
+                    IMDSV2_HEADER=(-H "X-aws-ec2-metadata-token: ${_token}")
+                  fi
+                  
+                  INSTANCE_ID=$(curl -s --retry 3 "${IMDSV2_HEADER[@]}" http://169.254.169.254/latest/meta-data/instance-id)
```

Comment:
What is this `169.254.169.254` ip exactly?

Reply:
[Retrieve instance metadata](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html)

Reply:
Seems like that IP address is a link-local IP address that is only accessible from within the instance and points to an instance metadata service that is running locally.

More details in the documentation: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html

Reply:
Ah didn't realize akartsky already replied (page wasn't refreshed)

Reply:
Thanks everyone for the details, it's clear now!